src/server/middleware/auth/authorizerbase.js

```diff
@@ -0,0 +1,122 @@
+/*globals*/
+/*jshint node:true*/
+/**
+ * @author pmeijer / https://github.com/pmeijer
+ */
+'use strict';
+
+var Q = require('q'),
+    ENTITY_TYPES = {
+        PROJECT: 'PROJECT',
+        USER: 'USER'
+    };
+
+/**
+ * @typedef {object} AccessRights
+ * @prop {boolean} read
+ * @prop {boolean} write
+ * @prop {boolean} delete
+ *
+ * @example
+ * {
+ *   read: true,
+ *   write: true,
+ *   delete: false
+ * }
+ */
+
+
+/**
+ *
+ * @param {GmeLogger} mainLogger
+ * @param {GmeConfig} gmeConfig
+ * @constructor
+ */
+function AuthorizerBase(mainLogger, gmeConfig) {
+
+    /**
+     * @type {{PROJECT: string, USER: string}}
+     */
+    this.ENTITY_TYPES = ENTITY_TYPES;
+
+    /**
+     * @type {GmeConfig}
+     */
+    this.logger = mainLogger.fork('AuthorizerBase');
+
+    /**
+     * @type {GmeLogger}
+     */
+    this.gmeConfig = gmeConfig;
+}
+
+/**
+ * @type {{PROJECT: string, USER: string}}
+ */
+AuthorizerBase.ENTITY_TYPES = ENTITY_TYPES;
+
+/**
+ *
+ * @param {object} params
+ * @param {object} params.collection - Mongo collection to default database.
```

Comment:
If it would be hairy to 'create' the collection object similar to gmeAuth inside the default authorizer, than it can stay as we can go around it (just ignoring it...), but eventually I think we should avoid this 'constraint'.